server/app/controllers/applicant/ApplicantProgramReviewController.java

```diff
@@ -106,12 +117,41 @@ public CompletionStage<Result> submit(Request request, long applicantId, long pr
                 if (cause instanceof SecurityException) {
                   return unauthorized();
                 }
+                if (cause instanceof ApplicationOutOfDateException) {
+                  Call reviewPage =
+                      routes.ApplicantProgramReviewController.review(applicantId, programId);
+                  return redirect(reviewPage).flashing("error", cause.getMessage());
+                }
                 throw new RuntimeException(cause);
               }
               throw new RuntimeException(ex);
             });
   }
 
+  /**
+   * Validates that the application is complete and correct to submit.
+   *
+   * <p>An application may be submitted but incomplete for a variety of reason:
```

Comment:
suggestion (blocking): The comment says "a variety of reasons" but then only lists one. Suggest combining into a single paragraph without the ul

Reply:
done